### PR TITLE
48 migrate to ethersjs

### LINKED ISSUE #48: Investigate: migrate to a different web3 library

### PR DESCRIPTION
**Pull request by felixwatts**
resolves #48 

- [x] Migrate from web3js to ethersjs
- [x] Migrate from truffle to waffle (includes migrate tests to typescript)
- [x] Update solidity compiler and solidity to 0.6.2

### CODE REVIEW

File: src/proofPointRegistry.ts [hunk 1]
```diff
@@ -317,52 +340,76 @@ class ProofPointRegistry {
...
+    return allEvents;
+  }
 
-    return events;
+  private async getEventsByFilter(
+    filter: ethers.EventFilter,
+    proofPointId: ProofPointId
+  ): Promise<ProofPointEvent[]> {
+    // TODO
+    // There seems to be a bug in ethers that you cannot specify the from and to block
```
Comment (pmackay) [inline src/proofPointRegistry.ts:396]: Could be handy to see if any issue raised on this....

File: test/ContractProofPointRegistry_v2.test.ts [hunk 2]
```diff
@@ -0,0 +1,342 @@
...
+    } catch (e) {}
+
+    // old registry cannot be used to commit
+    try {
+      await v1User1.commit(pp4);
+      expect.fail();
+    } catch (e) {}
+  });
+
+  it("version", async () => {
```
Comment (pmackay) [inline test/ContractProofPointRegistry_v2.test.ts:315]: Better test description?

**Review by pmackay — APPROVED** [review]
Looks cleaner, nice work!

File: src/proofPointRegistry.ts [hunk 3]
```diff
@@ -317,52 +340,76 @@ class ProofPointRegistry {
     proofPointId: ProofPointId
   ): Promise<ProofPointEvent[]> {
     // Prepare and store proxy object for the logic contract
-    const registry = new this._web3.eth.Contract(
-      ProofPointRegistryAbi[version].abi as any,
+    const registry = new ethers.Contract(
```
[the remark this answers is not in the document]
Comment (felixwatts) [reply src/proofPointRegistry.ts:343]: No, its only used as a namespace for types

File: src/proofPointRegistry.ts [hunk 4]
```diff
@@ -317,52 +340,76 @@ class ProofPointRegistry {
...
     );
 
-    const eventsRaw = await registry.getPastEvents("allEvents", {
-      // TODO filter doesn't work for some reason
-      // filter: {_claim: Web3.utils.keccak256(proofPointId) },
-      fromBlock: 0,
-      toBlock: "latest",
-    });
+    const filterIssued = registry.filters.Issued(
+      null,
```
[the remark this answers is not in the document]
Comment (felixwatts) [reply src/proofPointRegistry.ts:350]: Done

File: src/proofPointRegistry.ts (hunk 1 above)
Comment (felixwatts) [reply src/proofPointRegistry.ts:396]: Couldnt find one so raised one ethers-io/ethers.js#1148

File: test/ContractProofPointRegistry_v2.test.ts (hunk 2 above)
Comment (felixwatts) [reply test/ContractProofPointRegistry_v2.test.ts:315]: Done

File: src/proofPointRegistry.ts (hunk 1 above)
Comment (felixwatts) [reply src/proofPointRegistry.ts]: Fixed

**Pull request merged**